lib/xcode_summary/plugin.rb

```diff
@@ -183,16 +183,20 @@ def errors(action)
         Result.new(format_warning(result), result.location)
       end
 
-      test_failures = [
-        action.action_result.issues.test_failure_summaries,
-        action.build_result.issues.test_failure_summaries
-      ].flatten.compact.map do |summary|
-        result = Result.new(summary.message, parse_location(summary.document_location_in_creating_workspace))
-        Result.new(format_test_failure(result, summary.producing_target, summary.test_case_name),
+      if action.action_result.status != 'succeeded' then
```

Comment:
I'm not sure if this fully fixes the issue though. For example, If I have 2 tests, one which passed on 2nd iteration and another which failed all iterations, then this would post 2 test failures to the PR instead of just 1. Could lead to confusing results?

Reply:
@jwelton you are 100% right, this really only fixes the case where the overall action succeeds with after a retry.  Looking further into this, I'm wondering if this needs to be handled in the xcresult gem itself...

Reply:
This looks to be the area where `trainer` handles it.
https://github.com/fastlane/fastlane/blob/3d2d208c7cbec41b03e2e6ceed93590dbf90629f/trainer/lib/trainer/test_parser.rb#L295